include/thread_pool/worker.hpp

```diff
@@ -365,18 +331,18 @@ inline void Worker<Task, Queue>::threadFunc(const size_t id, WorkerVector& worke
             // We were unable to obtain a task. 
             // We now transition into the busy wait state.
             task_found = false;
-            num_busy_waiters.fetch_add(1, std::memory_order_acq_rel);
+            state->numBusyWaiters().fetch_add(1, std::memory_order_acq_rel);
```

Comment:
would it be useful to cache a reference to the result of `numBusyWaiters()` so that we don't have to access their values through the reference every time they are called? or is the compiler smart enough to optimize that away?